### PR TITLE
backend/x11: Give X11 a real rendering loop

### PR DESCRIPTION
**Pull request by ascent12**
This is some work split out of one of my renderer v6 PRs.

Makes use of the present extension to get notified of vsync, and not require any stupid timer hacks. Also make use of the present version of ConfigureNotify, because why not?

### CODE REVIEW

File: backend/x11/output.c [hunk 1]
```diff
@@ -217,22 +187,61 @@ struct wlr_output *wlr_x11_output_create(struct wlr_backend *backend) {
 	wlr_signal_emit_safe(&x11->backend.events.new_input, &output->pointer_dev);
 	wlr_signal_emit_safe(&x11->backend.events.new_input, &output->touch_dev);
 
+	// Start the rendering loop
+	wlr_output_send_frame(&output->wlr_output);
```
Comment (emersion) [inline backend/x11/output.c]: I don't think we need to do this. The compositor will damage the output and call `wlr_output_schedule_frame`.

**Comment by emersion** [comment]
This sounds like a good idea.

I get a black screen when running this on Sway with `WLR_BACKENDS=x11`.

**Comment by ascent12** [comment]
>I get a black screen when running this on Sway with WLR_BACKENDS=x11.

Weird. All of the wlroots examples I tried were working fine.
I'll have to look into that more.

**Comment by emersion** [comment]
FWIW, I get a black screen with both my updated rootston fork and sway.

**Comment by ascent12** [comment]
I figured it out. I was forgetting to call `wlr_output_update_enabled`, which `wlr_output_damage` pays attention to, but the naive examples do not.

File: backend/x11/output.c (hunk 1 above)
Comment (ascent12) [reply backend/x11/output.c]: Done

**Comment by emersion** [comment]
Thanks!

**Pull request merged**